oak_restricted_kernel/src/mm/page_tables.rs

```diff
@@ -45,10 +45,7 @@ pub unsafe fn create_offset_map<S: PageSize, A: FrameAllocator<Size4KiB>, M: Map
                 Page::<S>::from_start_address(offset + i * (S::SIZE as usize)).unwrap(),
                 frame,
                 flags,
-                PageTableFlags::PRESENT
-                    | PageTableFlags::GLOBAL
-                    | PageTableFlags::WRITABLE
-                    | PageTableFlags::ENCRYPTED,
+                PageTableFlags::PRESENT | PageTableFlags::WRITABLE | PageTableFlags::ENCRYPTED,
```

Comment:
If I were you I would add almost the entire commit message as a comment around here. Our future selves (or our successors) will thank us.

Reply:
Better yet, I found out what the difference is.

- AMD64 Architecture Programmer's Manual, Volume 2, Section 5.3: in PML4E (and PML5E), the `G` bit must be zero. `G` is ignored in PDPE and PDE (assuming 4K pages).
- Intel 64 and IA-32 Architectures Software Developer’s Manual, Volume 3A, Section 4.5: in `G` bit is ignored in all of PML5E, PML4E, PDPE and PDE (again, assuming 4K pages).

So there's a genuine CPU manufacturer difference here in how they handle page tables that caught me out.

I'll add a comment about this warning that you shouldn't set `GLOBAL` in the parent page table entries, as at best it'll get ignored and at worst it will cause failures.